### PR TITLE
integration step 1 and 2 for anomaly localization

### PR DESCRIPTION
**Pull request by wnbts**
Signed-off-by: lai <57818076+wnbts@users.noreply.github.com>

### Description
the changes are step [1](https://github.com/opensearch-project/ml-commons/blob/main/docs/how-to-add-new-function.md#step-1-name-the-function-1) and [2](https://github.com/opensearch-project/ml-commons/blob/main/docs/how-to-add-new-function.md#step-2-add-input-class-1) for anomaly localization integration.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Input.java [hunk 1]
```diff
@@ -12,19 +12,121 @@
...
+                    break;
+                case FIELD_ANOMALY_START_TIME:
+                    parser.nextToken();
+                    anomalyStartTime = Optional.of(parser.longValue());
+                    break;
+                case FIELD_FILTER_QUERY:
+                    ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                    filterQuery = Optional.of(parseInnerQueryBuilder(parser));
+                    ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.currentToken(), parser);
+                    break;
```
Comment (ylwu-amzn) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Input.java:124]: Skip children for default case?
```
                default:
                    parser.skipChildren();
                    break;
```

Comment (wnbts) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Input.java]: sure added

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Input.java [hunk 2]
```diff
@@ -12,19 +12,121 @@
...
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.index.query.AbstractQueryBuilder.parseInnerQueryBuilder;
+
 /**
  * Information about aggregate, time, etc to localize.
  */
 @Data
-public class Input {
+@AllArgsConstructor
+public class Input implements org.opensearch.ml.common.parameter.Input {
```
[the remark this answers is not in the document]
Comment (wnbts) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/Input.java]: ok, renamed.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Review by jackiehanyang — APPROVED** [review]

**Pull request merged**